### PR TITLE
Making CI config more generic

### PR DESCRIPTION
**Pull request by wdanilo**
### Pull Request Description
This PR implements new CI configuration:
- It is generalized, all sub-steps are written in a reusable way (in JavaScript, the GitHub YAML is generated).
- The CI configuration is now much more optimized and uses job dependencies (less resource hungry, faster to build).
- The CI supports now automatic releases and tags creation based on commits to branches `unstable` and `stable` and the entries in `CHANGELOG.md`. See docs (or below) for more information.
- The CDN integration is done.
- There is a new option in CI. If you want to trigger the building of artifacts on your local branch, just add `[ci build]` to your commit message!

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [x] All code has been manually tested in the "debug/interface" scene.

### CODE REVIEW

File: build/run.js [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
+const path          = require('path')
+const paths         = require('./paths')
+const prettier      = require("prettier")
+const stream        = require('stream')
+const yargs         = require('yargs')
+const zlib          = require('zlib')
 const child_process = require('child_process')
+const release       = require('./release')
+const workflow      = require('./workflow')
+const { promisify } = require('util')
```
Comment (farmaazon) [inline build/run.js:16]: Can we sort those "imports" alphabetically?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -227,3 +227,35 @@ have prepared several scripts which maximally automate the process:
...
+  `[major].[minor].[patch]-[sfx]`, where `[sfx]` is one of `alpha.[n]`, `beta.[n]`, or `rc.[n]`, 
+  where `[n]` is an unstable build number.
+
+- **stable**
+  Contains only those commits which can be considered stable product releases. Each commit to this
+  branch will result in a stable release of the product and will be published on GitHub as a 
+  release. The build version and build description will be automatically fetched from the newest 
+  `CHANGELOG.md` entry and will fail if the version will not be of the form 
+  `[major].[minor].[patch]`.
+
```
Comment (farmaazon) [inline docs/CONTRIBUTING.md:261]: This does not tell the full story how the CHANGELOG file should look like: That each version is a single-`#` section with the necessary "Enso " prefix and the version as described.

Do I understand correctly, that once we do merge to stable branch we modify changelog to merge all alphas, betas and rcs into a single changelog for a new stable version?

**Review by farmaazon — CHANGES_REQUESTED** [review]
* I like this JS code, it is very clear.
* Currently we don't make the release build on the feature branches, and thus we don't check the wasm binary size limit before merging to master (blocker)
* I think we also should check on feature branches if CHANGELOG file have a proper format.
* I would also add a new check to the PR template to make sure the CHANGELOG was updated if applicable.

**Comment by wdanilo** [comment]
> * I like this JS code, it is very clear.
Thanks!

> * Currently we don't make the release build on the feature branches, and thus we don't check the wasm binary size limit before merging to master. We should do it on at least one platform (blocker).
Good point, done.

> * I think we also should check on feature branches if CHANGELOG file have a proper format.
We do it on every execution of the `./run` script to update our JS configs, so it's already done.

> * I would also add a new check to the PR template to make sure the CHANGELOG was updated if applicable.
Done!

File: build/release.js [hunk 3]
```diff
@@ -86,7 +86,7 @@ function changelogEntries() {
     let entries  = []
     for (let section of sections) {
         if (!section.startsWith(prefix)) {
-            throw `Improper changelog entry header: ${section}`
+            throw `Improper changelog entry header: ${section}. It should start with`
```
Comment (farmaazon) [inline build/release.js:89]: Looks like something is missing.

File: docs/README_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+[79272]: http://github.com/ticket
+[79273]: http://github.com/ticket
+[79274]: http://github.com/ticket
+[79275]: http://github.com/ticket
+
+
+
+<br/>
+# Enso 47.0.0-alpha.8 (2049-01-24)
+...
```
Comment (farmaazon) [inline docs/README_TEMPLATE.md:163]: This file is hilarious :D however I wonder if we need so long a template?

**Review by farmaazon — APPROVED** [review]
Only minor comment - to consider, no need to be applied

**Pull request merged**